File: 2023/roadToZero.py
import sys


def threeSwimmers(lines):
	testCases = int(lines[0])
	for i in range(1, testCases + 1):
		line1 = lines[i]
		p, a, b, c = map(int, line1.split())
		if p % a == 0 or p % b == 0 or p % c == 0:
			print(0)  #print statement
		else:
			minV = min(a - (p % a), b - (p % b), c - (p % c))
			print(minV)  #print statement


lines = []
for line in sys.stdin:
	lines.append(line.strip())
threeSwimmers(lines)

File: 2023/js/roadToZero.js
"use strict";
exports.__esModule = true;
var readline_1 = require("readline");
function roadToZero(lines) {
	var testCases = Number(lines[0]);
	var line1;
	var line2;
	for (var i = 1; i <= testCases * 2; i++) {
		line1 = lines[i];
		line2 = lines[++i];
		var _a = line1.split(" ").map(Number),
			x = _a[0],
			y = _a[1];
		var _b = line2.split(" ").map(Number),
			a = _b[0],
			b = _b[1];
		var cost = 0;
		var minOfXnY = Math.min(x, y);
		if (2 * a > b) {
			x -= minOfXnY;
			y -= minOfXnY;
			cost += b * minOfXnY;
			var maxOfXnY = Math.max(x, y);
			// console.log(maxOfXnY);
			cost += maxOfXnY * a;
		} else {
			cost = a * x + b * y;
		}
		console.log(cost);
	}
}
var rl = readline_1.createInterface(process.stdin, process.stdout);
var input = [];
rl.on("line", function (line) {
	return input.push(line);
}).on("close", function () {
	roadToZero(input);
});
